### PR TITLE
Fix  ControlledPhaseShift

### PR DESCRIPTION
**Pull request by ryanlevy**
**Context:**

In PR #1064 the new ControlledPhaseShift was added, however the CNOT wires are fixed to 0 and 1 instead of the user input

**Description of the Change:** Wires in the CNOT decomposition are no longer fixed to 0 and 1
**Benefits:** Fixes new operation
**Possible Drawbacks:** None
**Related GitHub Issues:** None?

### CODE REVIEW

**Comment by josh146** [comment]
Wow! Great catch @ryanlevy. Thanks for this!

This is something that definitely should have been picked up by our tests, so a sign that our tests are lacking 🙂 Do you think you will be able to add a test that ensures the decomposition is correct for a weird set of wires (e.g., `wires=[2, 0]`)?

It could be something as simple as

```python
original_mat = qml.ControlledPhaseShift(x, wires=[2, 0]).matrix
decomposed_mats = [op.matrix for op in qml.ControlledPhaseShift.decomposition(x, wires=[2, 0])]
```
and performing matrix multiplication to ensure that the original matrix and decomposed matrix match.

**Comment by josh146** [comment]
By the way, don't forget to update the changelog with a link back to this PR, and add your name to the contributors section :)

**Comment by ryanlevy** [comment]
Thanks for the reminder! 

I've updated the tests, but the `.matrix` is a bit lacking to really test things (CNOT's matrix is the same if `wires=[0,1]` or `wires=[1,0]` for example). Hope this is ok!

**Comment by josh146** [comment]
> CNOT's matrix is the same if wires=[0,1] or wires=[1,0] for example

Ah, of course! This is because PennyLane allows arbitrary labels for wires (e.g., `CNOT(wires=['a', 0])`), and as a consequence of this, wire labels have no implicit ordering.

File: .github/CHANGELOG.md [hunk 1]
```diff
@@ -237,6 +237,9 @@ random_mat2 = rng.standard_normal(3, requires_grad=False)
 
 <h3>Bug fixes</h3>
 
+* Fixed ControlledPhaseShift/CPhase CNOT wires.
```
Comment (josh146) [inline .github/CHANGELOG.md:240]:
```suggestion
* Fixes incorrect wires in the decomposition of the `ControlledPhaseShift` operation.
```

**Review by josh146 — APPROVED** [review]
Looks great from my end, thanks @ryanlevy!
